network-config-analyzer/CanonicalIntervalSet.py

```diff
@@ -230,27 +231,96 @@ def is_subset(self, other):
             """
             return other.start <= self.start and other.end >= self.end
 
+    def find_interval_left(self, interval):
```

Comment:
Both `find_interval_left()` and `find_interval_right()` assume self is not empty, and will crash otherwise.
Better handle this case or otherwise document this assumption.
Also, `contained_in()` doesn't check if `other` is empty before calling `other.find_interval_left()`

Reply:
right.. fixed so that these methods will handle this case